src/renderer/features/contacts/EditRouteGuard/ui/EditRouteGuard.tsx

```diff
@@ -0,0 +1,26 @@
+import { ReactNode, useEffect } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { useStore } from 'effector-react';
+
+import * as editGuardModel from '../model/edit-guard';
+import { Contact } from '@renderer/entities/contact';
+
+type Props = {
+  redirectPath: string;
+  children?: ReactNode | ((contact: Contact) => ReactNode);
+};
+export const EditRouteGuard = ({ redirectPath, children }: Props) => {
```

Comment:
`Route guard` that redirects to address book if contact with id in query params doesn't exist

Reply:
Is there way to make this component and model universal?